snowpack/src/commands/dev.ts

```diff
@@ -292,14 +292,7 @@ export async function startServer(
 
   for (const [mountKey, mountEntry] of Object.entries(config.mount)) {
     logger.debug(`Mounting directory: '${mountKey}' as URL '${mountEntry.url}'`);
-    const files = glob.sync(path.join(mountKey, '**'), {
-      absolute: true,
-      nodir: true,
-      ignore: [
-        ...config.exclude,
-        ...(process.env.NODE_ENV === 'test' ? [] : config.testOptions.files),
-      ],
-    });
+    const files = (await new fdir().withFullPaths().crawl(mountKey).withPromise()) as string[];
```

Comment:
This isn't respecting files that should be ignored. I got it kind of got it working with this but there seems to still be a few straggler files that should be ignored. 

```tsx
  const excludeGlobs = [
    ...config.exclude,
    ...(process.env.NODE_ENV === 'test' ? [] : config.testOptions.files),
  ];
  const foundExcludeMatch = picomatch(excludeGlobs);
  
  const files = (await new fdir()
    .withFullPaths()
    .crawlWithOptions(mountKey, {
      includeBasePath: true,
      exclude: (_, dirPath) => {
        return foundExcludeMatch(dirPath);
      },
    })
    .withPromise()) as string[];

  ```

Reply:
I actually intentionally didn't add an ignored files check here to speed things up, since there's no harm in adding ignored files to the `fileToUrlMapping` map. Are you seeing issues with ignored files being handled here?

You can see us ignoring excluded files successfully in `snowpack/src/scan-imports.ts`

Reply:
> Are you seeing issues with ignored files being handled here?

No, just assumed it was needed since it was passing ignore properties before. 

> You can see us ignoring excluded files successfully in snowpack/src/scan-imports.ts

Ah yes, nice.

Reply:
yea, good call, I'll add a comment 👍